release: v1.45.0 (#4962)
apollographql/router: .changesets/feat_garypen_2002_subgraph_batching.md

Previously, the router preserved the concept of a batch until a `RouterRequest` finished processing. From that point, the router converted each batch request item into a separate `SupergraphRequest`, and the router planned and executed those requests concurrently within the router, then reassembled them into a batch of `RouterResponse` to return to the client. Now with the implementation in this release, the concept of a batch is extended so that batches are issued to configured subgraphs (all or named). Each batch request item is planned and executed separately, but the queries issued to subgraphs are optimally assembled into batches which observe the query constraints of the various batch items.

To configure subgraph batching, you can enable `batching.subgraph.all` for all subgraphs. You can also enable batching per subgraph with `batching.subgraph.subgraphs.*`. For example:

```yaml
batching:
  enabled: true
  mode: batch_http_link
  subgraph:
    # Enable batching on all subgraphs
    all:
      enabled: true
```

```yaml
batching:
  enabled: true
  mode: batch_http_link
  subgraph:
    # Disable batching on all subgraphs
    all:
      enabled: false

>     # Configure(over-ride) batching support per subgraph

Reviewer: ```suggestion
    # Configure (override) batching support per subgraph
```

    subgraphs:
      subgraph_1:
        enabled: true
      subgraph_2:
        enabled: true
```

> Note: `all` may be over-ridden by `subgraphs`. This applies in general for all router subgraph configuration options.

Reviewer: ```suggestion
Note: `all` can be overridden by `subgraphs`. This applies in general for all router subgraph configuration options.
```
